nmdc_server/storage.py

```diff
@@ -0,0 +1,217 @@
+import re
+from datetime import datetime, timedelta, timezone
+from enum import StrEnum
+from functools import cached_property
+from pathlib import PurePath
+from typing import Any, Iterable
+
+from google.auth.credentials import AnonymousCredentials
+from google.cloud.exceptions import NotFound
+from google.cloud.storage import Blob, Bucket, Client
+
+from nmdc_server.config import settings
+from nmdc_server.schemas import SignedUrl
+
+
+class BucketName(StrEnum):
+    """Enum for GCS bucket names"""
+
+    SUBMISSION_IMAGES = "nmdc-submission-images"
+
+
+class Storage:
+    """A class to manage Google Cloud Storage interactions."""
+
+    def __init__(self, project_id: str | None, use_fake_server: bool):
```

Comment:
I suggest adding a docstring (caution: I drafted this one on my iPhone screen 🔬 )

```suggestion
    def __init__(self, project_id: str | None, use_fake_server: bool):
    """
    :param project_id: Google Cloud Project ID
    :param use_fake_server: Whether you want to use a fake GCS service or the real one
    """
```